src/main/java/online/book/store/service/BookServiceImpl.java

```diff
@@ -34,4 +34,19 @@ public BookDto findById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id).orElseThrow(() ->
                 new EntityNotFoundException("Can't find book by id: " + id)));
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        if (bookRepository.existsById(id)) {
+            Book book = bookMapper.toModel(requestDto);
+            book.setId(id);
+            return bookMapper.toDto(bookRepository.save(book));
+        }
+        throw new EntityNotFoundException("Can't find book with id: " + id);
```

Comment:
let's not make the 'default' flow of the update method were it throws an exception. flip this `if` so that validation checks are structured like this:
```java
if (somethingWrong) {
    throw ex;
}
// another check
// business logic and other 'normal' flow of the program
```
this way you can stack up validations one **after** another and not one **inside** another, this is also called a [return early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8)